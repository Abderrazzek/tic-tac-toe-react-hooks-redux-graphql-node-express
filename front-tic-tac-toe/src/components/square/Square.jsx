import React from "react";

const Square = ({ value, onClick }) => {
  const style = value ? `squares ${value}` : `squares`;

  return (
    <button data-testid="square" className={style} onClick={onClick}>
      {value}
    </button>
  );
};

export default Square;
