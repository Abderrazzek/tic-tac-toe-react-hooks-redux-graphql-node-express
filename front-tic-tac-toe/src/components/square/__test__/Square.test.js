import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import ReactDom from "react-dom";
import React from "react";

import Square from "../Square";

//cleanup after each element rendering
afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDom.render(<Square />, div);
});

it("renders Square correctly", () => {
  const { getByTestId } = render(<Square value="X" onClick={() => {}} />);
  expect(getByTestId("square")).toHaveTextContent("X");
});

it("matches snapshot", () => {
  const component = renderer.create(<Square value="X" onClick={() => {}} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
