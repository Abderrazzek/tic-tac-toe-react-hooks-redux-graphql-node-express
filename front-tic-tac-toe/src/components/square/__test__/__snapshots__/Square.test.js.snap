// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="squares X"
  data-testid="square"
  onClick={[Function]}
>
  X
</button>
`;

exports[`matches snapshot 2`] = `
<button
  className="squares X"
  data-testid="square"
  onClick={[Function]}
>
  X
</button>
`;
